crypto/evp/digest.c

```diff
@@ -517,11 +516,7 @@ int EVP_DigestFinalXOF(EVP_MD_CTX *ctx, unsigned char *md, size_t size)
         return 0;
     }
 
-    params[i++] = OSSL_PARAM_construct_size_t(OSSL_DIGEST_PARAM_XOFLEN, &size);
-    params[i++] = OSSL_PARAM_construct_end();
-
-    if (EVP_MD_CTX_set_params(ctx, params) > 0)
-        ret = ctx->digest->dfinal(ctx->algctx, md, &size, size);
+    ret = ctx->digest->dfinal(ctx->algctx, md, &size, size);
```

Comment:
This is problematic for backwards compatibility. A third party provider might expect that the param is set. So I would keep setting the param in EVP_DigestFinalXOF.

Reply:
In that case, please add a comment explaining that, 'cause otherwise, it looks weirdly unnecessary!

Reply:
Done